src/main/scala/shine/OpenCL/Compilation/HostCodeCodeGenerator.scala

```diff
@@ -1,24 +1,25 @@
-package shine.OpenCL
+package shine.OpenCL.Compilation
 
 import arithexpr.arithmetic
 import shine.C
 import shine.C.AST._
+import shine.C.Compilation.CodeGenerator
 import shine.DPIA.Phrases._
 import shine.DPIA.Types._
 import shine.DPIA.VarType
-import shine.OpenCL.compilation.HostManagedBuffers
 import shine.OpenCL.primitives.imperative.{HostExecution, KernelCallCmd, NewManagedBuffer}
+import shine.OpenCL._
 
 import scala.collection.{immutable, mutable}
 
-object HostCodeGenerator {
-  def apply(): HostCodeGenerator =
-    new HostCodeGenerator(mutable.ListBuffer[Decl](), immutable.Map[String, arithmetic.Range]())
+object HostCodeCodeGenerator {
```

Comment:
This is very much deliberate.
We generate `HostCode`, therefore, it is a `(HostCode)(CodeGenerator)`.
But, I would be ok reverting this (I think at some point I wanted to avoid a clash with the `HostModuleGenerator` which had a different name at some point ...
What do you think?

Reply:
I prefer `HostCodeGenerator`, it is not a code generator of host code, it is simply a generator of host code.